### PR TITLE
Request breakpointPositions for all identical sources (#2482)

### PR DESCRIPTION
**Pull request by hbenl**

### CODE REVIEW

File: src/devtools/client/debugger/src/actions/breakpoints/index.js [hunk 1]
```diff
@@ -198,7 +199,25 @@ export function updateHoveredLineNumber(line) {
     // Set the initial location here as a placeholder to be checked after any async activity.
     dispatch(actions.setHoveredLineNumberLocation(initialLocation));
 
-    await dispatch(setBreakpointPositions({ sourceId: source.id, line }));
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/actions/breakpoints/index.js:201]: This feels like something where we either want to update `setBreakpointPositions` to always do this for all of the sources that match the URL or add a new `setBreakpointPositionsForURL`.

File: src/devtools/client/debugger/src/actions/breakpoints/index.js [hunk 2]
```diff
@@ -198,7 +199,25 @@ export function updateHoveredLineNumber(line) {
...
+    const promises = [];
+    promises.push(dispatch(setBreakpointPositions({ sourceId: source.id, line })));
+    if (source.url) {
+      // there may be multiple identical sources for the same URL (due to navigations),
+      // in this case we need to request breakpoint positions for all of them so that
+      // we can set logpoints in all of them later on
+      const sourceIds = ThreadFront.getSourceIdsForURL(source.url);
+      // sourceIds will only contain "preferred" sources, check if the current source
+      // is one of them (otherwise it's not identical to them)
+      if (sourceIds.contains(source.id)) {
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/actions/breakpoints/index.js:211]: when is this not the case?

File: src/devtools/client/debugger/src/actions/breakpoints/index.js [hunk 3]
```diff
@@ -198,7 +199,25 @@ export function updateHoveredLineNumber(line) {
...
+        for (sourceId of sourceIds) {
+          if (sourceId !== source.id) {
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/actions/breakpoints/index.js:213]: I think we can probably simplify this by dropping the IF.


```js
if (source.url) {
  const sourceIds = ThreadFront.getSourceIdsForURL(source.url);
  return Promise.all(sourceIds.map(sourceId => Thingy(sourceId))
} else {
 return Thingy(sourceId)
}
```

**Review by jasonLaster — COMMENTED** [review]
Looking good. Can we also add some end-to-end tests that cover these navigation cases?

**Comment by hbenl** [comment]
Superseded by #2628

**Pull request closed without merging**